component-catalog-connectors/artifactory-connector/README.md

```diff
@@ -0,0 +1,203 @@
+
+### Install from PyPI
+
+  ```
+  $ pip install elyra-artifactory-catalog-connector
+  ```
+
+### Install from source code
+
+   ```
+   $ git clone https://github.com/elyra-ai/examples.git
+   $ cd examples/component-catalog-connectors/artifactory-connector/
+   $ make source-install
+   ```
+
+## Use the connector
+
+### Add a new Artifactory catalog
+
+1. Launch JupyterLab
+2. Open the  [`Manage Components` panel](https://elyra.readthedocs.io/en/stable/user_guide/pipeline-components.html#managing-custom-components-using-the-jupyterlab-ui)
+3. Click `+` then `New Artifactory Component Catalog`
+4. Specify a "name" for the catalog
+5. (Optional) Specify a "category name" under which the catalog's components will be organized in the palette
+6. Specify the required "Source" parameters:
+    - `artifactory_url`
+    - `repository_name`
+    - `repository_path`
+    - `max_recursion_depth`
+    - `max_files_per_folder`
+    - `file_filter`
+    - `file_ordering`
+7. (Optional) Specify the optional "Source" parameters:
+    - `artifactory_username`
+    - `artifactory_password`
+8. Save the catalog entry
+9. Open the Visual Pipeline Editor and expand the palette. The catalog components are displayed.
+
+### Recommended Usage
+
+We recommend that you version your component YAML rather than having users always point to the latest version of each component.
+Otherwise, when changes are made to `inputs`/`outputs`/`implementation`, existing pipelines may break or do unexpected things.
+
+> 🟦 __Tip__ 🟦
+>
+> Include the version in each component's `name` so they can be distinguished in the Elyra UI.
+>
+> For example:
+> ```yaml
+> name: My Component - 1.0.0
+> description: ""
+> inputs: []
+> outputs: []
+> implementation: {}
+> ```
+
+Elyra only stores pointers to the source-catalog of each component in its `.pipeline` files.
+This means your catalogs must continue to include ALL old component versions, otherwise old pipelines will be unable to run.
+
+You can solve this problem with the `Artifactory Catalog Connector` by adding TWO catalog instances for your single Artifactory repo:
+
+1. that has only the latest version of each component:
+    - `max_files_per_folder = 1`
+    - `file_ordering = VERSION_DESCENDING`
+2. that has all versions of each component:
+    - `max_files_per_folder = -1`
+
+> 🟦 __Tip__ 🟦
+>
+> Name your component YAML files with consistent prefixes before the version.
+> 
+> This is because `VERSION_DESCENDING` treats the whole file-name as a version.
+> For example, `aaaa-1.0.0.yaml` would be sorted before `bbbb-9.0.0.yaml`.
+>
+> Alternatively, don't include any prefix on file names, for example `1.0.0.yaml`, `1.1.0.yaml`.
+
+
+### Example Configs
+
+Assume an artifactory server `https://example.com/artifactory/` has a repository called `elyra-components` 
+with the following folder structure:
+
+```
+component_1/
+   __COMPONENT__
```

Comment:
We should introduce the purpose of these special files much earlier, since they are fundamental to how the proposed implementation works. 

Have you considered using a single metadata file that would, for example, live in the root directory and contain information that governs whether component files in a specific location are processed? @lresende, what's your take?

Reply:
I could use some of what I wrote in https://github.com/elyra-ai/elyra/issues/2139#issuecomment-1144571300, while explaining the behavior in the issue for a GitHub component catalog.

But I agree I can make those docs more clear, I will fix it.

Reply:
@ptitzler I have significantly improved the readme, please tell me if its good enough now.

Reply:
Yep! It's much more clear now! Thank you!